pkg/hooks/read.go

```diff
@@ -0,0 +1,83 @@
+// Package hooks implements CRI-O's hook handling.
+package hooks
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	current "github.com/projectatomic/libpod/pkg/hooks/1.0.0"
+	"github.com/pkg/errors"
+)
+
+type reader func(content []byte) (*current.Hook, error)
+
+var (
+	// ErrNoJSONSuffix represents hook-add attempts where the filename
+	// does not end in '.json'.
+	ErrNoJSONSuffix = errors.New("hook filename does not end in '.json'")
+
+	// Readers registers per-version hook readers.
+	Readers = map[string]reader{}
+)
+
+// Read reads a hook JSON file, verifies it, and returns the hook configuration.
+func Read(path string) (*current.Hook, error) {
+	if !strings.HasSuffix(path, ".json") {
+		return nil, ErrNoJSONSuffix
+	}
+	content, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	hook, err := read(content)
+	if err != nil {
+		return nil, errors.Wrapf(err, "parsing hook %q", path)
+	}
+	err = hook.Validate()
+	return hook, err
+}
+
+func read(content []byte) (hook *current.Hook, err error) {
+	var ver version
+	if err := json.Unmarshal(content, &ver); err != nil {
+		return nil, errors.Wrap(err, "version check")
+	}
+	reader, ok := Readers[ver.Version]
+	if !ok {
+		return nil, fmt.Errorf("unrecognized hook version: %q", ver.Version)
```

Comment:
Do existing hooks have a version?  I don't think so, we would need to warn and set the version to 1.0.

Reply:
> Do existing hooks have a version? I don't think so, we would need to warn and set the version to 1.0.

I'm retroactively declaring them to be 0.1.0, see the initial comment here, [these docs][1], and [this code][2].

[1]: https://github.com/wking/libpod/blob/1fef0077b473d778f6d1b9797afa81036d57a9a2/pkg/hooks/README.md#010-hook-schema
[2]: https://github.com/projectatomic/libpod/pull/686/files#diff-ee156200100589cfe6afdf31e9a81731R90